src/parcsr_ls/par_ilu.c

```diff
@@ -12,140 +12,114 @@
+   hypre_ParILUDataSchurGMRESMaxIter(ilu_data)            = 5;
+   hypre_ParILUDataSchurGMRESTol(ilu_data)                = 0.0;
+   hypre_ParILUDataSchurGMRESAbsoluteTol(ilu_data)        = 0.0;
+   hypre_ParILUDataSchurGMRESRelChange(ilu_data)          = 0;
    /* schur precond data */
-   hypre_ParILUDataSchurPrecondIluType(ilu_data) = 0;
-   hypre_ParILUDataSchurPrecondIluLfil(ilu_data) = 0;
-   hypre_ParILUDataSchurPrecondIluMaxRowNnz(ilu_data) = 100;
-   hypre_ParILUDataSchurPrecondIluDroptol(ilu_data) =
-      NULL;/* this is not the default option, set it only when switched to */
-   hypre_ParILUDataSchurPrecondPrintLevel(ilu_data) = 0;
-   hypre_ParILUDataSchurPrecondMaxIter(ilu_data) = 1;
-   hypre_ParILUDataSchurPrecondTriSolve(ilu_data) = 1;
+   hypre_ParILUDataSchurPrecondIluType(ilu_data)          = 0;
+   hypre_ParILUDataSchurPrecondIluLfil(ilu_data)          = 0;
+   hypre_ParILUDataSchurPrecondIluMaxRowNnz(ilu_data)     = 100;
+   /* this is not the default option, set it only when switched to */
+   hypre_ParILUDataSchurPrecondIluDroptol(ilu_data)       = NULL;
+   hypre_ParILUDataSchurPrecondPrintLevel(ilu_data)       = 0;
+   hypre_ParILUDataSchurPrecondMaxIter(ilu_data)          = 1;
+   hypre_ParILUDataSchurPrecondTriSolve(ilu_data)         = 1;
    hypre_ParILUDataSchurPrecondLowerJacobiIters(ilu_data) = 5;
    hypre_ParILUDataSchurPrecondUpperJacobiIters(ilu_data) = 5;
-   hypre_ParILUDataSchurPrecondTol(ilu_data) = 0.0;
-
+   hypre_ParILUDataSchurPrecondTol(ilu_data)              = 0.0;
    /* -> SCHUR-NSH */
-   hypre_ParILUDataSchurNSHSolveMaxIter(ilu_data) = 5;
-   hypre_ParILUDataSchurNSHSolveTol(ilu_data) = 0.0;
-   hypre_ParILUDataSchurNSHDroptol(ilu_data) =
-      NULL;/* this is not the default option, set it only when switched to */
-
-   hypre_ParILUDataSchurNSHMaxNumIter(ilu_data) = 2;
-   hypre_ParILUDataSchurNSHMaxRowNnz(ilu_data) = 1000;
-   hypre_ParILUDataSchurNSHTol(ilu_data) = 1e-09;
-
-   hypre_ParILUDataSchurMRMaxIter(ilu_data) = 2;
-   hypre_ParILUDataSchurMRColVersion(ilu_data) = 0;
-   hypre_ParILUDataSchurMRMaxRowNnz(ilu_data) = 200;
-   hypre_ParILUDataSchurMRTol(ilu_data) = 1e-09;
+   hypre_ParILUDataSchurNSHSolveMaxIter(ilu_data)         = 5;
+   hypre_ParILUDataSchurNSHSolveTol(ilu_data)             = 0.0;
+   /* this is not the default option, set it only when switched to */
+   hypre_ParILUDataSchurNSHDroptol(ilu_data)              = NULL;
+   hypre_ParILUDataSchurNSHMaxNumIter(ilu_data)           = 2;
+   hypre_ParILUDataSchurNSHMaxRowNnz(ilu_data)            = 1000;
+   hypre_ParILUDataSchurNSHTol(ilu_data)                  = 1e-09;
+   hypre_ParILUDataSchurMRMaxIter(ilu_data)               = 2;
+   hypre_ParILUDataSchurMRColVersion(ilu_data)            = 0;
+   hypre_ParILUDataSchurMRMaxRowNnz(ilu_data)             = 200;
+   hypre_ParILUDataSchurMRTol(ilu_data)                   = 1e-09;
 
-   return (void *)                        ilu_data;
+   return ilu_data;
```

Comment:
I think we need to cast `ilu_data` to `void *` in the return statement

Reply:
I don't know if we actually need. But it does seem more correct to cast. I didn't get warnings for that though.